src/components/markdown-htmlview.component.js

```diff
@@ -0,0 +1,335 @@
+
+import { ImageZoom } from 'components';
+import { colors, emojis, fonts, normalize } from 'config';
+
+const lightFont = {
+  ...fonts.fontPrimaryLight,
+};
+
+const regularFont = {
+  ...fonts.fontPrimary,
+};
+
+const textStyleLight = {
+  ...lightFont,
+  fontSize: Platform.OS === 'ios' ? normalize(11) : normalize(12),
+  color: colors.primaryDark,
+};
+
+const textStyleRegular = {
+  ...regularFont,
+  fontSize: Platform.OS === 'ios' ? normalize(11) : normalize(12),
+  color: colors.primaryDark,
+};
+
+const textStyle = Platform.OS === 'ios' ? textStyleLight : textStyleRegular;
+
+const boldStyle = {
+  ...textStyle,
+  ...fonts.fontPrimarySemiBold,
+};
+const italicStyle = { ...textStyle, ...fonts.fontPrimaryItalic };
+const hrStyle = { borderBottomWidth: 2, borderBottomColor: colors.greyDark };
+const underlineStyle = { textDecorationLine: 'underline' };
+const codeStyle = {
+  ...fonts.fontCode,
+  backgroundColor: colors.greyMidLight,
+  fontSize: normalize(9),
+};
+const linkStyle = { ...boldStyle, color: colors.primaryDark };
+
+const styles = {
+  b: boldStyle,
+  strong: boldStyle,
+  i: italicStyle,
+  em: italicStyle,
+  u: underlineStyle,
+  pre: { ...codeStyle, padding: 5, marginBottom: 10 },
+  code: codeStyle,
+  span: textStyle,
+  p: { ...textStyle, margin: 0, padding: 0 },
+  h1: {
+    ...fonts.fontPrimarySemiBold,
+    ...hrStyle,
+    fontSize: normalize(24),
+  },
+  h2: {
+    ...fonts.fontPrimarySemiBold,
+    ...hrStyle,
+    fontSize: normalize(20),
+  },
+  h3: { ...fonts.fontPrimarySemiBold, fontSize: normalize(18) },
+  h4: { ...fonts.fontPrimarySemiBold, fontSize: normalize(16) },
+  h5: { ...fonts.fontPrimarySemiBold, fontSize: normalize(14) },
+  h6: { ...fonts.fontPrimarySemiBold, fontSize: normalize(12) },
+  hr: { ...hrStyle },
+  li: textStyle,
+  a: linkStyle,
+};
+
+const stylessheet = StyleSheet.create(styles);
+
+marked.setOptions({
+  renderer: new marked.Renderer(),
+  gfm: true,
+  tables: true,
+  breaks: false,
+  pedantic: false,
+  sanitize: false,
+  smartLists: true,
+  smartypants: false,
+});
+
+const { height, width } = Dimensions.get('window');
+
+const headingRenderer = (node, index, siblings, parent, defaultRenderer) => {
+  const style = node.name === 'h1' || node.name === 'h2' ? { ...hrStyle } : {};
+
+  return (
+    <View key={index} style={style}>
+      <Text style={{ ...styles[node.name] }}>
+        {defaultRenderer(node.children, node)}
+      </Text>
+    </View>
+  );
+};
+
+export class MarkdownHtmlview extends Component {
+  props: {
+    source: String,
+    onLinkPress: Function,
+  };
+
+  transformMarkdown = md => {
+    const issueReference = /(\s)#(\d+)/;
+    const profileReference = /(\s)@([_0-9A-Za-z]+)/;
+    const todoItem = /<li>\s*\[(x|\s)?\](.*)<\/li>/g;
+    const emojiMarkup = /:(\w+):/g;
+
+    let rendered = marked(md);
+
+    rendered = rendered
+      .replace(new RegExp(/<p>*>/g), '<span>')
```

Comment:
Hey, should be better use Just `/<p>*>/g`, I think `new RegExp` is not necessary :)